### DISCUSSION TITLE
wal2json for historical logs

### DISCUSSION THREAD

**Discussion by jacksonhurst** [eulerto/wal2json#241] [Q&A] [answered]
Hello,

I have a unique situation where I want to stream database changes from the past 3 months to a different database (not postgres). 

Is there any functionality within wal2json or a similar tool that I can simply indicate a XID or LSN and it will dig through the WALs until it finds that point, then stream changes from that point onward?

I have found tons of solutions for streaming new changes and for replicating an entire database, but nothing in the realm of reading back and parsing through historical WALs.

Any ideas?

**Comment by eulerto** [comment, marked as answer]
I'm afraid it is not possible. If you have created a replication slot 3 months ago, you could use it to consume the desired time frame.
